### PR TITLE
[otp_ctrl] Increase LFSR reseeding period

### PR DESCRIPTION
**Pull request by msfschaffner**
The OTP background checks do not happen frequently (only every few seconds). Hence, the LFSR does not have to be reseeded very frequently, either.

This changes the reseed counter mechanism to be usage-based.
The LFSR is only reseeded once 16 values have been drawn from it.

Signed-off-by: Michael Schaffner <msf@opentitan.org>

### CODE REVIEW

**Comment by sriyerg** [comment]
I'm not happy about making changes to the design after certifying D2V2. D2 means "FEATURE_FROZEN", which clearly is not in this case. Could we hold off on merging this PR at least until after we have a proper, well documented strategy / flow in mind?

**Comment by msfschaffner** [comment]
Sorry I should explain this change better. 
The currently implemented reseeding mechanism is complete from a functional perspective, but I had some concerns that it will be drawing too much entropy from EDN without good reason.

I.e., with the current reseeding counter of 2^16, this would be constantly drawing 64 bit * 25 MHz / 2^16 =  24.4kbit/s, which seems wasteful, given that the background check timers will be set to values in the order of seconds in practice. 

A switch to usage-based reseeding as implemented in this PR would address this in a dynamic way (i.e., adjust based on the current background timer configuration). But given that OTP is in D2/V2 it would probably be better to minimize the RTL diff and just increase the width of the reseeding counter from 16bit to 256bit, effectively reducing the entropy consumption to around 100bit/s with a one line change only:

https://github.com/lowRISC/opentitan/blob/ff34c73ae8209e0e060c94428c8c2aadec5357f1/hw/ip/otp_ctrl/rtl/otp_ctrl_lfsr_timer.sv#L36

This should not have an adverse effect on coverage except that it may take longer in simulation until the EDN interface is exercised. If that is a concern, one possible solution would be to just force the parameter to 16bit or less in simulation.

**Review by cindychip — COMMENTED** [review]
Changes LGTM, maybe let other designers to take a look before merging?

**Review by cindychip — APPROVED** [review]

**Pull request merged**
